Tests/DataStreamTests.swift

```diff
@@ -0,0 +1,749 @@
+        var response: HTTPURLResponse?
+        var streamOnMain = false
+        var completeOnMain = false
+        let expect = expectation(description: "stream should complete")
+
+        // When
+        AF.streamRequest("https://httpbin.org/bytes/\(expectedSize)").responseStream { stream in
+            switch stream.event {
+            case let .stream(result):
+                streamOnMain = Thread.isMainThread
+                switch result {
+                case let .success(data):
+                    accumulatedData.append(data)
+                }
+            case let .complete(completion):
+                completeOnMain = Thread.isMainThread
+                response = completion.response
+                expect.fulfill()
+            }
+        }
+
+        waitForExpectations(timeout: timeout)
+
+        // Then
+        XCTAssertEqual(response?.statusCode, 200)
+        XCTAssertEqual(accumulatedData.count, expectedSize)
+        XCTAssertTrue(streamOnMain)
+        XCTAssertTrue(completeOnMain)
+    }
+
+    func testThatDataCanBeStreamedManyTimes() {
+        // Given
+        let expectedSize = 1000
+        var firstAccumulatedData = Data()
+        var firstResponse: HTTPURLResponse?
+        var firstStreamOnMain = false
+        var firstCompleteOnMain = false
+        let firstExpectation = expectation(description: "first stream should complete")
+        var secondAccumulatedData = Data()
+        var secondResponse: HTTPURLResponse?
+        var secondStreamOnMain = false
+        var secondCompleteOnMain = false
+        let secondExpectation = expectation(description: "second stream should complete")
+
+        // When
+        AF.streamRequest(URLRequest.makeHTTPBinRequest(path: "bytes/\(expectedSize)"))
+            .responseStream { stream in
+                switch stream.event {
+                case let .stream(result):
+                    firstStreamOnMain = Thread.isMainThread
+                    switch result {
+                    case let .success(data):
+                        firstAccumulatedData.append(data)
+                    }
+                case let .complete(completion):
+                    firstCompleteOnMain = Thread.isMainThread
+                    firstResponse = completion.response
+                    firstExpectation.fulfill()
+                }
+            }
+            .responseStream { stream in
+                switch stream.event {
+                case let .stream(result):
+                    secondStreamOnMain = Thread.isMainThread
+                    switch result {
+                    case let .success(data):
+                        secondAccumulatedData.append(data)
+                    }
+                case let .complete(completion):
+                    secondCompleteOnMain = Thread.isMainThread
+                    secondResponse = completion.response
+                    secondExpectation.fulfill()
+                }
+            }
```

Comment:
Pretty cool that this is supported. I'm not sure what an actual use case would be, but I'm sure someone will do something really cool with it.